application/src/main/java/org/togetherjava/tjbot/logging/discord/DiscordLogForwarder.java

```diff
@@ -108,14 +108,9 @@ void forwardLogEvent(LogEvent event) {
 
     private void processPendingLogs() {
         try {
-            if (rateLimitExpiresAt != null) {
-                if (Instant.now().isBefore(rateLimitExpiresAt)) {
-                    // Still rate limited, try again later
-                    return;
-                }
-
-                // Rate limit has expired, try again
-                rateLimitExpiresAt = null;
+            if (handleIsRateLimitActive()) {
```

Comment:
like the method extraction but maybe the name `isRateLimitActive` is better suited 😄

Reply:
we use this `handle` prefix a lot in the codebase, the intention is to make it clear that the method does more than a simple `is/has` check.

in this case, there is a side effect with `rateLimitExpiresAt` which you wouldn't expect from a method that _checks if the rate limit is active_ (and nothing else).

Reply:
I agree that it reads weird, especially if you are not used to it, but it's the convention we went with. And it's better to be a bit weird, than to be a liar, I guess ^^

Reply:
yeah. i couldnt come up with a better name, but `isRateLimitActive()` would be misleading and thats dangerous. but having the method vs not having it at all is definitely better, it did improve the code readability.

maybe someone else has a better name idea. it does:
* `isRateLimitActive`
* but additionally maintains the internal state for this.
point is, it must be called every now and then. if people just check the field themselves instead of using the method, they will see outdated information.